pkg/alertmanager/alertmanager_metrics.go

```diff
@@ -47,8 +47,8 @@ type alertmanagerMetrics struct {
 	silencesPropagatedMessagesTotal *prometheus.Desc
 }
 
-func newAlertmanagerMetrics() *alertmanagerMetrics {
-	return &alertmanagerMetrics{
+func newAlertmanagerMetrics(r prometheus.Registerer) *alertmanagerMetrics {
```

Comment:
`alertmanagerMetrics` implements `prometheus.Collector` interface. `Collectors` typically don't register into registry themselves, but are registered by client code using them. I think we should keep that, and not pass `registerer` into `newAlertmanagerMetrics`.